src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -99,56 +100,89 @@ private static int[] ComputeKMPFailureFunction(byte[] pattern)
         }
 
         // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
-        private static int KMPSearch(byte[] pattern, byte[] bytes)
+        private static unsafe int KMPSearch(byte[] pattern, MemoryMappedViewAccessor accessor)
         {
             int m = 0;
             int i = 0;
             int[] table = ComputeKMPFailureFunction(pattern);
 
-            while (m + i < bytes.Length)
+            byte* bytes = null;
+            accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref bytes);
```

Comment:
Still missing PointerOffset: https://github.com/dotnet/sdk/pull/2293#discussion_r192828196

My memory is actually unclear on when it is going to be non-zero, but it bit us in metadata reader.

Reply:
Sorry, this should be right https://github.com/dotnet/sdk/pull/2293/files#diff-b3b03fee742dfb740c837b74d3d814bbR145